tests/src/test/scala/cats/tests/ReducibleTests.scala

```diff
@@ -0,0 +1,20 @@
+package cats
+package tests
+
+class ReducibleTestsAdditional extends CatsSuite {
+
+  test("Reducible[NonEmptyList].reduceM stack safety") {
+    def nonzero(acc: Long, x: Long): Option[Long] =
+      if (x == 0) None else Some(acc + x)
+
+    val n = 100000L
+    val expected = n*(n+1)/2
+    (1L to n).toList.toNel.foreach { nel =>
+      val actual = nel.reduceM(Option.apply)(nonzero)
+      assert(actual.get == expected)
+    }
+  }
```

Comment:
This should be fine since `(1L to n).toList.toNel` should never return `None`, but I think I'd rather either see it explicitly use `Option.get` or do something like:

```scala
val actual = (1L to n).toList.toNel.flatMap(_.reduceM(Option.apply)(nonzero))
actual should === (Some(expected))
```